Sources/SpelledPitch/PitchSpeller/Wetherfield/SpellingInverter.swift

```diff
@@ -0,0 +1,430 @@
+        self.flowNetwork = DirectedGraph(internalNodes: internalNodes(spellings: spellings))
+        self.pitchClass = { int in spellings[int]?.pitchClass }
+        
+        let specificEdgeScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> = (upDownEdgeScheme.pullback(nodeMapper) + sameEdgeScheme.pullback(nodeMapper))
+            * connectDifferentInts
+        
+        let sameIntEdgesScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sameIntsScheme * connectSameInts
+        
+        let specificSourceScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sourceEdgeLookupScheme.pullback(nodeMapper)
+        
+        let specificSinkScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sinkEdgeLookupScheme.pullback(nodeMapper)
+        
+        let allSchemes: [DirectedGraphScheme<PitchSpeller.UnassignedNode>] = [
+            specificEdgeScheme,
+            sameIntEdgesScheme,
+            specificSourceScheme,
+            specificSinkScheme
+        ]
+        
+        let maskScheme: DirectedGraphScheme<PitchSpeller.AssignedNode> =
+            allSchemes.reduce(DirectedGraphScheme { _ in false }, +)
+                .pullback({ $0.unassigned })
+        
+        // Apply masking of specific manifestations of the general global adjacency schemes
+        self.flowNetwork.mask(maskScheme)
+    }
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Computed Properties
+    
+    /// - Returns: A concrete distribution of weights to satisfy the weight relationships delimited by
+    /// `weightDependencies` or `nil` if no such distribution is possible, i.e. there are cyclical
+    /// dependencies between edge types. In the latter case, the spellings fed in are *inconsistent*.
+    /// Weights are parametrized by `Pitch.Class` and `Tendency` values.
+    var weights: [PitchedEdge: Double] {
+        
+        let dependencies = pitchedDependencies
+        precondition(!findCycle(dependencies))
+        
+        func dependeciesReducer (
+            _ weights: inout [PitchedEdge: Double],
+            _ dependency: (key: PitchedEdge, value: Set<PitchedEdge>)
+            ) {
+            
+            func recursiveReducer (
+                _ weights: inout [PitchedEdge: Double],
+                _ dependency: (key: PitchedEdge, value: Set<PitchedEdge>)
+                ) -> Double {
+                let weight = dependency.value.reduce(1.0) { result, edge in
+                    if weights[edge] != nil { return result + weights[edge]! }
+                    return result + recursiveReducer(&weights, (key: edge, value: dependencies[edge]!))
+                }
+                weights[dependency.key] = weight
+                return weight
+            }
+            
+            let _ = recursiveReducer(&weights, dependency)
+        }
+        
+        return dependencies.reduce(into: [PitchedEdge: Double](), dependeciesReducer)
```

Comment:
Not sure if this is that helpful, but you can do:

```Swift
return dependencies.reduce(into: [:], dependeciesReducer)
```

The types are inferred just fine.